connectors/saavn.js

```diff
@@ -0,0 +1,125 @@
+/**
+ * Chrome-Last.fm-Scrobbler - Saavn.com Connector
+ *
+ * Original Author: Mayank Meghwanshi [mayankmeghwanshi@gmail.com]
+ * Testing, Fixes, Refactor: Shadab Zafar [dufferzafar0@gmail.com]
+ *
+ * Saavn.com's now playing bar does not have Artist's name
+ * but there is a link to the Track's information - so we make an
+ * Ajax call and extract artist information.
+ */
+
+// DOM Nodes to keep track for song details
+var DOM_TRACK_NAME = '#player-track-name';
+var DOM_TRACK_DURATION = '#track-time';
+
+Connector.playerSelector = '#now-playing';
+
+// Prevent same track from being scrobbled continuously
+var previousTrack = '';
+var previousUrl = '';
+
+// Get track name
+function getTrack() {
+    var trackName = $(DOM_TRACK_NAME + ' a').text();
+    if (!trackName) {
+        return ;
+    }
+
+    return trackName;
+}
+
+// Get track length
+function getDuration() {
+    if (!$(DOM_TRACK_DURATION)) {
+        return;
+    }
+
+    var duration = $(DOM_TRACK_DURATION).text().split(':');
+    return 60 * parseInt(duration[0]) + parseInt(duration[1]);
+}
+
+function updateNowPlaying() {
+
+    var track = getTrack();
+    var duration = getDuration();
+
+    // Extract Track's URL to find artist information
+    var trackURL = $(DOM_TRACK_NAME + ' a').attr("onclick");
```

Comment:
There's a link to the album page, yes, which includes the information but there's no indicator as to which song is now playing, so we'll have to find that from the list, right?